core/server/api/v2/webhooks.js

```diff
@@ -0,0 +1,47 @@
+const models = require('../../models');
+const common = require('../../lib/common');
+
+module.exports = {
+    docName: 'webhooks',
+    add: {
+        statusCode: 201,
+        headers: {},
+        options: [
+            'include'
+        ],
+        permissions: true,
+        query(frame) {
+            return models.Webhook.getByEventAndTarget(
+                frame.data.webhooks[0].event,
+                frame.data.webhooks[0].target_url,
+                frame.options
+            ).then((webhook) => {
+                if (webhook) {
+                    return Promise.reject(
+                        new common.errors.ValidationError({message: common.i18n.t('errors.api.webhooks.webhookAlreadyExists')})
+                    );
+                }
+
+                return models.Webhook.add(frame.data.webhooks[0], frame.options);
+            });
+        }
+    },
+    destroy: {
+        statusCode: 204,
+        headers: {},
+        options: [
+            'id'
+        ],
+        validation: {
+            options: {
+                id: {
+                    required: true
+                }
+            }
+        },
+        permissions: true,
+        query(frame) {
+            return models.Webhook.destroy(frame.options).return(null);
```

Comment:
This will not 404 correctly if the webhook doesn't exist

Reply:
`frame.options.require=true`

Reply:
🤔 Hmm, I thought that got handled implicitly, as v0.1 didn't have anything extra for that.

Reply:
Yeah v0.1 did not use require, but it should.
It's not handled implicit right now :)